src/components/views/dialogs/spotlight/SpotlightDialog.tsx

```diff
@@ -1065,6 +1065,56 @@ const SpotlightDialog: React.FC<IProps> = ({ initialText = "", initialFilter = n
             </>
         );
     }
+    // calculate no of element to jump based on arrow key or page up and page down
+    const getNoOfEleToJump = ({
+        currentIndex,
+        totalElements,
+        noOfJump,
+    }: {
+        currentIndex: number;
+        totalElements: number;
+        noOfJump: number;
+    }) => {
+        if (currentIndex + noOfJump < 0) {
+            return 0;
+        } else if (currentIndex + noOfJump > totalElements - 1) {
+            return totalElements - 1;
+        } else {
+            return currentIndex + noOfJump;
+        }
+    };
```

Comment:
This could be replaced by `Math.max(0, Math.min(ref.length - 1, idx + offset))`, it's just clamping after all

Reply:
We also have a `clamp` utility method

Reply:
In that case, please use that clamp method :)